src/pkg/registry/client.go

```diff
@@ -93,6 +93,8 @@ type Client interface {
 	// is used to specify whether the destination artifact will be overridden if
 	// its name is same with source but digest isn't
 	Copy(srcRepository, srcReference, dstRepository, dstReference string, override bool) (err error)
+	// Do the request
+	Do(req *http.Request) (*http.Response, error)
```

Comment:
It seems a common HTTP request handle function, is this the right place to add this func? 
It's very different with other functions in this interface.

Reply:
yes, I have to add this in order to call it in adapter

Reply:
@bitsf @ywk253100 

Let's raise a debt issue to track the refactor work for the future.